contracts/handlers/GenericHandlerV1.sol

```diff
@@ -0,0 +1,112 @@
+          executeContractAddress       address   padded to 32 bytes   bytes 64  - 96
+          maxFee:                      uint256                        bytes 96  - 128
+          metaData:
+            metadataDepositor:         address   padded to 32 bytes   bytes 128 - 156
+            executionData:             bytes                          bytes 160 - END
+     */
+    function deposit(bytes32 resourceID, address depositor, bytes calldata data) external returns (bytes memory) {
+        require(data.length > 160, "Incorrect data length");
+
+        uint256        lenMetadata;
+        uint256        metadataDepositor;
+        bytes   memory metaData;
+
+        lenMetadata = abi.decode(data, (uint256));
+        metaData = bytes(data[128:128 + lenMetadata]);
+
+        assembly {
+            metadataDepositor := mload(add(metaData, 44))
+        }
+        // metaData contains:       0x + depositorAddress + executionData************************
+        // Shift it 12 bytes right: 0x000000000000000000000000depositorAddress
+        require(depositor == address(uint160(metadataDepositor >> 96)), 'incorrect depositor in deposit data');
+    }
+
+    /**
+        @notice Proposal execution should be initiated when a proposal is finalized in the Bridge contract.
+        @param resourceID ResourceID used to find address of contract to be used for deposit.
+        @param data Structure should be constructed as follows:
+          len(metaData):               uint256                        bytes 0   - 32
+          executeFuntionSignature:     bytes4    padded to 32 bytes   bytes 32  - 64
+          executeContractAddress       address   padded to 32 bytes   bytes 64  - 96
+          maxFee:                      uint256                        bytes 96  - 128
+          metaData:
+            metadataDepositor:         address   padded to 32 bytes   bytes 128 - 160
+            executionData:             bytes                          bytes 160 - END
+     */
+    function executeProposal(bytes32 resourceID, bytes calldata data) external {
+        uint256        lenMetadata;
+        bytes32        executeFuntionSignature;
+        uint256        executeContractAddress;
+        uint256        maxFee;
+        bytes   memory executionData;
+        bytes4         functionSignature;
+        address        contractAddress;
+
+        (lenMetadata, executeFuntionSignature, executeContractAddress, maxFee) = abi.decode(data, (uint256, bytes32, uint256, uint256));
+
+        functionSignature = bytes4(bytes(data[60:64]));
+        executionData          = bytes(data[160:128 + lenMetadata]);
+        contractAddress   = address(uint160(executeContractAddress));
+
+        bytes memory callData = abi.encodePacked(functionSignature, executionData);
+        (bool success, ) = contractAddress.call(callData);
+        require(success, "delegatecall to contractAddress failed");
```

Comment:
I would try/catch this. A fail here is not considered a fail of proposal execution.

Reply:
This will trigger try/catch on `Bridge` contract and log this error message. It's not directly a fail of execution, but informs you of the fail reason.

Reply:
You're right, but try/catch can only be used for external function calls and contract creation calls.
Any way this requires changes :sweat_smile:

Reply:
We don't even need try/catch probably we just don't need the require :smile: